contracts/zero-ex/contracts/src/features/LimitOrdersFeature.sol

```diff
@@ -0,0 +1,875 @@
+        {
+            address signer = LibSignature.getSignerOfHash(
+                orderInfo.orderHash,
+                params.signature
+            );
+            if (signer != params.order.maker) {
+                LibLimitOrdersRichErrors.OrderNotSignedByMakerError(
+                    orderInfo.orderHash,
+                    signer,
+                    params.order.maker
+                ).rrevert();
+            }
+        }
+
+        // Pay the protocol fee.
+        (ethProtocolFeePaid, ) = _collectProtocolFee(params.order.pool, params.taker);
+
+        // Settle between the maker and taker.
+        (takerTokenFilledAmount, makerTokenFilledAmount) = _settleOrder(
+            SettleOrderInfo({
+                orderHash: orderInfo.orderHash,
+                maker: params.order.maker,
+                taker: params.taker,
+                makerToken: IERC20TokenV06(params.order.makerToken),
+                takerToken: IERC20TokenV06(params.order.takerToken),
+                makerAmount: params.order.makerAmount,
+                takerAmount: params.order.takerAmount,
+                takerTokenFillAmount: params.takerTokenFillAmount,
+                takerTokenFilledAmount: orderInfo.takerTokenFilledAmount
+            })
+        );
+
+        // Pay the fee recipient.
+        if (params.order.takerTokenFeeAmount > 0) {
+            uint256 feeAmount = LibMathV06.getPartialAmountFloor(
+                takerTokenFilledAmount,
+                params.order.takerAmount,
+                params.order.takerTokenFeeAmount
+            );
+            LibTokenSpender.spendERC20Tokens(
+                params.order.takerToken,
+                params.taker,
+                params.order.feeRecipient,
+                feeAmount
+            );
+        }
+
+        emit LimitOrderFilled(
+            orderInfo.orderHash,
+            params.order.maker,
+            params.taker,
+            params.order.feeRecipient,
+            address(params.order.makerToken),
+            address(params.order.takerToken),
+            takerTokenFilledAmount,
+            makerTokenFilledAmount,
+            params.order.pool
+        );
+    }
+
+    /// @dev Fill an RFQ order. Private variant. Does not refund protocol fees.
+    /// @param order The RFQ order.
+    /// @param signature The order signature.
+    /// @param takerTokenFillAmount Maximum taker token to fill this order with.
+    /// @param taker The order taker.
+    /// @return ethProtocolFeePaid How much protocol fee was paid.
+    /// @return takerTokenFilledAmount How much maker token was filled.
+    /// @return makerTokenFilledAmount How much maker token was filled.
+    function _fillRfqOrderPrivate(
+        LibLimitOrder.RfqOrder memory order,
+        LibSignature.Signature memory signature,
+        uint128 takerTokenFillAmount,
+        address taker
+    )
+        private
+        returns (
+            uint256 ethProtocolFeePaid,
```

Comment:
why do we bubble up the `ethProtocolFeePaid` but not the `wethProtocolFeePaid` for these functions?

Reply:
Just because the weth one isn't useful for our accounting right now.

Reply:
Maybe if we decide to emit the protocol fee paid as part of the event? We should discuss this in the v4 meeting.